apps/emqx_management/src/emqx_mgmt_api_api_keys.erl

```diff
@@ -218,10 +220,38 @@ api_key_by_name(put, #{bindings := #{name := Name}, body := Body}) ->
     Enable = maps:get(<<"enable">>, Body, undefined),
     ExpiredAt = ensure_expired_at(Body),
     Desc = maps:get(<<"desc">>, Body, undefined),
-    case emqx_mgmt_auth:update(Name, Enable, ExpiredAt, Desc) of
-        {ok, App} -> {200, emqx_mgmt_auth:format(App)};
-        {error, not_found} -> {404, ?NOT_FOUND_RESPONSE}
+    Role = maps:get(<<"role">>, Body, ?ROLE_API_DEFAULT),
+    case emqx_mgmt_auth:update(Name, Enable, ExpiredAt, Desc, Role) of
+        {ok, App} ->
+            {200, emqx_mgmt_auth:format(App)};
+        {error, not_found} ->
+            {404, ?NOT_FOUND_RESPONSE};
+        {error, Reason} ->
+            {400, #{
+                code => 'BAD_REQUEST',
+                message => iolist_to_binary(io_lib:format("~p", [Reason]))
+            }}
     end.
 
 ensure_expired_at(#{<<"expired_at">> := ExpiredAt}) when is_integer(ExpiredAt) -> ExpiredAt;
 ensure_expired_at(_) -> infinity.
+
+-if(?EMQX_RELEASE_EDITION == ee).
+
+app_extend_fields() ->
+    [
+        {role,
+            hoconsc:mk(binary(), #{
+                desc => ?DESC(role),
+                default => ?ROLE_API_DEFAULT,
+                example => ?ROLE_API_DEFAULT,
+                validator => fun emqx_dashboard_rbac:valid_api_role/1
+            })}
```

Comment:
Maybe replace binary as ENUM will be better.
- the API is more clear, and don't need validator.
- the role field is atom.

Reply:
It can't be an atom, we plan to allow users to create custom roles.